pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -910,18 +939,22 @@ public void destroy() {
 
     // Re-order documents using the inverted index
     RealtimeInvertedIndexReader invertedIndex = indexContainer._invertedIndex;
-    int[] docIds = new int[_numDocsIndexed];
+    int[] docIds = new int[numDocsIndexed];
+    int[] batch = new int[256];
```

Comment:
Not related, but good enhancement. Suggest putting it in a separate PR and we can directly merge?

Reply:
I'll revert this